relay-general/src/store/transactions/processor.rs

```diff
@@ -11,19 +11,27 @@ use crate::types::{Annotated, Meta, ProcessingAction, ProcessingResult, Remark,
 
 use super::TransactionNameRule;
 
+/// Configuration around removing high-cardinality parts of URL transactions.
+#[derive(Debug, Default)]
+pub struct TransactionNameConfig<'r> {
+    /// True if regex patterns should be applied to erase identifiers from the transaction name.
+    pub scrub_identifiers: bool,
+    /// True if transaction names scrubbed by regex patterns should be marked as [`TransactionSource::Sanitized`].
+    ///
+    /// Transaction names modified by clusterer rules are always marked as such.
+    pub mark_scrubbed_as_sanitized: bool,
+    /// Rules for identifier replacement that were discovered by Sentry's transaction clusterer.
+    pub rules: &'r [TransactionNameRule],
+}
+
 /// Rejects transactions based on required fields.
-#[derive(Default)]
 pub struct TransactionsProcessor<'r> {
-    normalize_names: bool,
-    tx_name_rules: &'r [TransactionNameRule],
+    name_config: &'r TransactionNameConfig<'r>,
 }
 
 impl<'r> TransactionsProcessor<'r> {
-    pub fn new(normalize_names: bool, tx_name_rules: &'r [TransactionNameRule]) -> Self {
-        Self {
-            normalize_names,
-            tx_name_rules,
-        }
+    pub fn new(name_config: &'r TransactionNameConfig<'r>) -> Self {
+        Self { name_config }
```

Comment:
Just suggestion, so if instead of providing reference here and implementing default with static name config, we could accept here a reference to `LightNormalizationConfig` and then build the config inside of the new? Like following:
```suggestion
    pub fn new(config: &'r LightNormalizationConfig) -> Self {
        let name_config = TransactionNameConfig {
            scrub_identifiers: config.transaction_name_config.scrub_identifiers,
            mark_scrubbed_as_sanitized: config.transaction_name_config.mark_scrubbed_as_sanitized,
            rules: config.transaction_name_config.rules,
        };
        Self { name_config }
  }

```

This way you could still `#[derive(Default)]` on `TransactionsProcessor` and in `normalize.rs` you could just do 
```rust
let mut transactions_processor =
            transactions::TransactionsProcessor::new(config);
```

And of course update tests to use new `new` function 😄 

Right now I'm not sure what would be better,  maybe having this the way it is now it's a bit clearer and have better separation of concerns.

Reply:
I have a better idea now, let's make `light_normalize_event` accept not a reference but a a moved variable of `LightNormalizationConfig` and then we could easily move `transaction_name_config` out of it and move it into `new` function. 
This way I think the code will code much cleaner and also there won't be additional jumps through the reference when that not really needed.

@jjbayer  Let me know what you think and we could pair on this.

Reply:
Good idea, done in https://github.com/getsentry/relay/pull/1917/commits/e2633be36ffdf09aead4955f507b778e5b41488c.